src/node_interface.rs

```diff
@@ -73,11 +73,11 @@ pub trait Interface : Sync + Send {
     /// an address for further handling of the request is provided. Failure is indicated as an
     /// InterfaceError.
     fn handle_post(&mut self,
-                   our_authority: Authority,
-                   from_authority: Authority,
-                   from_address: NameType,
-                   name : NameType,
-                   data: Data) -> Result<MessageAction, InterfaceError>;
+                   our_authority  : Authority,
+                   from_authority : Authority,
+                   from_address   : NameType,
```

Comment:
we can address on Monday whether from_address is needed, also name is in our_authority and in data itself